crates/mirror-mirror/src/set.rs

```diff
@@ -0,0 +1,38 @@
+use crate::Reflect;
+
+use core::fmt;
+
+pub trait Set: Reflect {
+    fn len(&self) -> usize;
+
+    fn is_empty(&self) -> bool;
+
+    fn try_insert(&mut self, element: &dyn Reflect) -> Result<bool, SetError>;
+
+    fn try_remove(&mut self, element: &dyn Reflect) -> Result<bool, SetError>;
+
+    fn try_contains(&mut self, element: &dyn Reflect) -> Result<bool, SetError>;
+
+    fn iter(&self) -> Iter<'_>;
```

Comment:
These feel like the core set operations to me. Not sure if including other iterators like difference, intersection, and union are worth (or possible)

Reply:
I would say let's just have these for now we can always add more if we find the need